gradle/intTestImage.gradle

```diff
@@ -32,16 +33,26 @@ if (useIncomingDistributions) {
 
     intTestImage {
         dependsOn unpackIncomingAllDistribution
-        from unpackedPath
+        from(unpackedPath) {
+            exclude "lib/gradle-version-info*.jar"
+        }
+        into("lib") {
+            from { project(':versionInfo').testVersionInfoJar }
+        }
```

Comment:
Then the tests, when using `GradleVersion`, would obtain a different result compared to what the distribution under test would obtain. We would then fail integration tests like [this](https://github.com/gradle/gradle/blob/881d52da9f0c9d55ec1b8a7fd42e1be201a9844d/subprojects/core/src/integTest/groovy/org/gradle/api/ApplyPluginIntegSpec.groovy#L134-L134).